File: examples/cpp/core/schema_view/schema_view.cpp
#include "schema_view_reflection.h"

#include <bond/core/bond.h>
#include <bond/stream/output_buffer.h>

using namespace examples::schema_view;


int main()
{
    Example example;
    
    example.num = 42;
    example.str = "test";
    example.items.push_back(3.14);
    example.items.push_back(0);
    
    bond::OutputBuffer output;
    bond::CompactBinaryWriter<bond::OutputBuffer> writer(output);

    Marshal(example, writer);

    ExampleView view;

    bond::InputBuffer input(output.GetBuffer());

    Unmarshal(input, view);
    
    BOOST_ASSERT(example.num == view.num);
    BOOST_ASSERT(example.str == view.str);
    
    return 0;    
}

File: examples/java/core/schema_view/src/main/java/com/microsoft/bond/examples/SchemaView.java
package com.microsoft.bond.examples;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.IOException;

import com.microsoft.bond.Deserializer;
import com.microsoft.bond.Serializer;

import com.microsoft.bond.protocol.CompactBinaryReader;
import com.microsoft.bond.protocol.CompactBinaryWriter;

import com.microsoft.bond.examples.schemaview.Example;
import com.microsoft.bond.examples.schemaview.ExampleView;

public class SchemaView {

    public static void main(final String[] args) throws IOException {

        final Example example = new Example();
        example.num = 42;
        example.str = "test";
        example.items.add(3.14D);
        example.items.add(0D);

        final ByteArrayOutputStream output = new ByteArrayOutputStream();
        final CompactBinaryWriter writer = new CompactBinaryWriter(output, (short) 1);

        final Serializer<Example> serializer = new Serializer<>();
        serializer.serialize(example, writer);

        final ByteArrayInputStream input = new ByteArrayInputStream(output.toByteArray());

        final CompactBinaryReader reader = new CompactBinaryReader(input, (short) 1);
        final Deserializer<ExampleView> deserializer = new Deserializer<>(ExampleView.BOND_TYPE);
        final ExampleView view = deserializer.deserialize(reader);

        assert example.num == view.num;
        assert example.str.equals(view.str);
    }
}
